rfd/000?-database-access.md

```diff
@@ -0,0 +1,512 @@
+by the user authority and re-encode all identity and routing information
+extracted from the certificate supplied by the client into it.
+
+### Database service <---> database instance
+
+**Mutual TLS (onprem).**  When connecting to an onprem database instance, the
+database service will use a client certificate for authentication, which it
+will generate on the fly according to the database requirements (for example,
+PostgreSQL requires the database user name to be encoded in the certificate
+as common name).
+
+This also means that the database needs to be configured with Teleport's
+certificate authority and (in general case) use server certificate issued by
+Teleport which can be done using `tctl auth sign` command (see below for UX).
+
+**Password auth (RDS/Aurora).** Amazon RDS/Aurora don't support client cert
+authentication. Instead, they support IAM authentication so Teleport database
+service will generate a short-lived authentication token using RDS API that
+will be used as a password:
+
+https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.IAMDBAuth.html
+
+In order to be able to verify RDS/Aurora server certificate, users will need
+to download RDS root certificate and configure the database service to trust
+it:
+
+https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/UsingWithRDS.SSL.html
+
+## Configuration
+
+### Teleleport database service
+
+The following new configuration section is added to the Teleport file config:
+
+```yaml
+# New global key housing the database service configuration.
+db_service:
+  # Enable or disable the database service.
+  enabled: "yes"
+  # List of the database this service is proxying.
+  databases:
+    # Database instance name, used to refer to an instance in CLI like tsh.
+  - name: "postgres-prod"
+    # Optional free-form verbose description of a database instance.
+    description: "Production instance of PostgreSQL 13.0"
+    # Database procotol, only "postgres" is supported initially.
+    protocol: "postgres"
+    # Database connection endpoint, should be reachable from the service.
+    endpoint: "postgres.internal.example.com:5432"
+    # Optional CA certificate path, e.g. for AWS RDS/Aurora.
+    ca_path: "/path/to/root.pem"
+    # AWS specific configuration for RDS/Aurora databases.
+    aws:
+      # Use IAM authentication with RDS/Aurora database.
+      auth: "iam"
+      # Optional AWS region RDS/Aurora database is running in.
+      region: "us-east-1"
+      # Optional AWS access key, will be taken from environment if unspecified.
+      access_key: "XXX"
+      # Optional AWS secret key, will be taken from environment if unspecified.
+      secret_key: "YYY"
+    # Static labels assigned to the database instance, used in RBAC.
+    labels:
+      env: "stage"
+    # Dynamic labels assigned to the database instance, used in RBAC.
+    commands:
+    - name: "time"
+      command: ["date", "+%H:%M:%S"]
+      period: "1m"
```

Comment:
For TAA and kube services, we renamed these to `static_labels` and `dynamic_labels`

Reply:
@awly @russjones is this change implemented for nodes as well? If not, please make sure it's global for all config files (and backwards compatible). Otherwise we will end up with different terminology for same concepts.

Reply:
Good point, this was not changed for `ssh_service`.
Filed https://github.com/gravitational/teleport/issues/4771